kai/reactive_codeplanner/task_manager/task_manager.py

```diff
@@ -295,6 +297,10 @@ def handle_new_tasks_after_processing(
             for t in similar_tasks:
                 unprocessed_new_tasks.remove(t)
             self.handle_ignored_task(task)
+            # Once we have a retry or an ignored task, we should wait to add
+            # children until the task is completed.
+            # On ignored task, we now revert to the before snapshot work
+            return
```

Comment:
just trying to understand, would this also stop yielding other tasks within the same depth level?

Reply:
no because this is in the handle_new_tasks_after_processing method, it stops doing more work there. The only work that is below this, is the attaching child tasks, which we don't want to do. The task_queue either gets the retried task back on it, or it is moved to ignored and the next task at the same depth will be taken.